File: translator/testFiles/javascript/cases/enclosingThis.js
{
    classes = function () {
        var Point = Class.create({initialize:function (x, y) {
            this.$x = x;
            this.$y = y;
        }, get_x:function () {
            return this.$x;
        }, get_y:function () {
            return this.$y;
        }, mul:function () {
            return (that = this, function (scalar) {
                return new Point(that.get_x() * scalar, that.get_y() * scalar);
            })
                ;
        }
        });
        return {Point:Point};
    }
        ();
    Anonymous = Namespace.create({initialize:function () {
        Anonymous.$m = (new Anonymous.Point(2, 3)).mul();
    }, get_m:function () {
        return Anonymous.$m;
    }, box:function () {
        var answer = Anonymous.get_m()(5);
        return answer.get_x() === 10 && answer.get_y() === 15 ? 'OK' : 'FAIL';
    }
    }, classes);
    Anonymous.initialize();
}

File: compiler/testData/codegen/box/closures/enclosingThis.kt
// IGNORE_BACKEND_FIR: JVM_IR
class Point(val x:Int, val y:Int) {
    fun mul() :  (scalar:Int)->Point  {
        return { scalar:Int -> Point(x * scalar, y * scalar) }
    }
}

val m = Point(2, 3).mul()

fun box() : String {
    val answer = m(5)
    return if (answer.x == 10 && answer.y == 15) "OK" else "FAIL"
}
